src/util/macros.h

```diff
@@ -30,7 +30,7 @@ inline char* C_STRING(v8::Local<v8::String> string) {
 
 // Given a double, returns whether the number is a valid 32-bit signed integer.
 inline bool IS_32BIT_INT(double num) {
-	return floor(num) == num && num < 2147483648 && num >= -2147483648;
+	return floor(num) == num && num < 2147483648L && num >= -2147483647L - 1;
```

Comment:
Did it not work to do this?

`return floor(num) == num && num < 2147483648.0 && num >= -2147483648.0;`

Looking back at it, I'd prefer this, since all of the type would be the same (`double`)